snapcraft/_options.py

```diff
@@ -177,14 +178,18 @@ def debug(self):
         return self.__debug
 
     def __init__(self, use_geoip=False, parallel_builds=True,
-                 target_deb_arch=None, debug=False):
+                 target_deb_arch=None, debug=False, cache_dir=None):
         # TODO: allow setting a different project dir and check for
         #       snapcraft.yaml
         self.__project_dir = os.getcwd()
         self.__use_geoip = use_geoip
         self.__parallel_builds = parallel_builds
         self._set_machine(target_deb_arch)
         self.__debug = debug
+        if cache_dir:
```

Comment:
Alright I removed this from `ProjectOptions` and simply made them optional keyword arguments in the relevant areas so the tests could hook into them.